cot/src/cache/stores/memory.rs

```diff
@@ -0,0 +1,243 @@
+//! In-memory cache store implementation.
+//!
+//! This module provides a simple thread-safe, process-local cache store that
+//! implements the generic [`CacheStore`] trait. It is primarily intended for
+//! development, testing, and low-concurrency scenarios where a shared in-memory
+//! map is sufficient.
+//!
+//! # Examples
+//!
+//! ```no_run
+//! use cot::cache::stores::memory::Memory;
+//! use cot::cache::stores::CacheStore;
+//! use serde_json::json;
+//!
+//! #[tokio::main]
+//! async fn main() {
+//!     let store = Memory::new();
+//!     let key = "example_key".to_string();
+//!     let value = json!({"data": 42});
+//!
+//!     store.insert(key.clone(), value, Default::default()).await.unwrap();
+//!     let retrieved = store.get(&key).await.unwrap();
+//!
+//!     assert_eq!(retrieved, Some(json!({"data": 42})));
+//! }
```

Comment:
```suggestion
//! # #[tokio::main]
//! # async fn main() {
//!     let store = Memory::new();
//!     let key = "example_key".to_string();
//!     let value = json!({"data": 42});
//!
//!     store.insert(key.clone(), value, Default::default()).await.unwrap();
//!     let retrieved = store.get(&key).await.unwrap();
//!
//!     assert_eq!(retrieved, Some(json!({"data": 42})));
//! # }
```

We can hide the `main` function stuff as it's not relevant for the docs. Similarly with other doctests.